src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NativeInteger.cs

```diff
@@ -916,6 +916,33 @@ public void AttributeFieldExists()
             });
         }
 
+        [Theory]
+        [InlineData("C<dynamic, nint, System.IntPtr>", "C<dynamic, System.IntPtr, System.IntPtr>", "False, False, True, False")]
+        [InlineData("C<dynamic, nuint, System.UIntPtr>", "C<dynamic, System.UIntPtr, System.UIntPtr>", "False, False, True, False")]
+        [InlineData("C<T, nint, System.IntPtr>", "C<T, System.IntPtr, System.IntPtr>", "False, False, True, False")]
+        [InlineData("C<T, nuint, System.UIntPtr>", "C<T, System.UIntPtr, System.UIntPtr>", "False, False, True, False")]
+        public void NestedNativeIntegerWithPrecedingType(string sourceType, string metadataType, string expectedAttribute)
+        {
+            var comp = CompileAndVerify($@"
+class C<T, U, V>
+{{
+    public {sourceType} F;
+}}
```

Comment:
Please consider writing out all the cases inline as separate fields in a single test.